src/envoy/common/metadata_object_test.cc

```diff
@@ -98,7 +92,6 @@ TEST_F(WorkloadMetadataObjectTest, FromBaggage) {
   EXPECT_EQ(gotDeploy->workloadName(), "foo");
```

Comment:
@douglas-reid this breaks hashing, doesn't it? there's is no instance for deployment.

Reply:
@howardjohn why isn't instance set for deployments?

Reply:
Where is it trying to read from? WorkloadMetadataResource over ECDS?

If so, as far as I can tell we should always have the instance name?

if its from baggage, then I don't think that is included as part of baggage, the format we have is just `k8s.deployment.name=<deployment_name>`